verification/src/genesis_verifier.rs

```diff
@@ -0,0 +1,213 @@
+use super::Verifier;
+use crate::{
+    error::CellbaseError, BlockErrorKind, BlockVerifier, EpochError, NumberError, UnclesError,
+    UnknownParentError,
+};
+use ckb_chain_spec::consensus::Consensus;
+use ckb_dao_utils::genesis_dao_data_with_satoshi_gift;
+use ckb_error::Error;
+use ckb_types::{core::BlockView, packed::CellInput};
+
+#[derive(Clone)]
+pub struct GenesisVerifier {}
+
+impl GenesisVerifier {
+    pub fn new() -> Self {
+        GenesisVerifier {}
+    }
+}
+
+impl Default for GenesisVerifier {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Verifier for GenesisVerifier {
+    type Target = Consensus;
+
+    fn verify(&self, consensus: &Self::Target) -> Result<(), Error> {
+        NumberVerifier::new().verify(consensus.genesis_block())?;
+        EpochVerifier::new().verify(consensus.genesis_block())?;
+        ParentHashVerifier::new().verify(consensus.genesis_block())?;
+        CellbaseVerifier::new().verify(consensus.genesis_block())?;
+        UnclesVerifier::new().verify(consensus.genesis_block())?;
+        DAOVerifier::new(consensus).verify(consensus.genesis_block())?;
+        BlockVerifier::new(consensus).verify(consensus.genesis_block())
+    }
+}
+
+#[derive(Clone)]
+pub struct NumberVerifier {}
+
+impl Default for NumberVerifier {
```

Comment:
Please remove all `Default` trait, you should only add it when needed

Reply:
This is only for eliminate the warning from clippy.

Reply:
`NumberVerifier` and other verifiers are empty struct, no need to impl `new` fn and `Default` trait, you can use it directly:
```rust
pub struct NumberVerifier;

impl NumberVerifier {
    pub fn verify(&self, block: &BlockView) -> Result<(), Error> {
        ...
    }
}

NumberVerifier.verify(consensus.genesis_block())?;
```

btw, I prefer static method to instance method, since these verifiers are non-contextual and empty structs.
```rust
pub struct NumberVerifier;

impl NumberVerifier {
    pub fn verify(block: &BlockView) -> Result<(), Error> {
        ...
    }
}

NumberVerifier::verify(consensus.genesis_block())?;
```

Reply:
You are right, updated.